File: remove_placeholders.js
const fs = require('fs');
const path = require('path');

function removePravatar(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            if (!fullPath.includes('build') && !fullPath.includes('.dart_tool') && !fullPath.includes('.git')) {
                removePravatar(fullPath);
            }
        } else if (file.endsWith('.dart')) {
            let content = fs.readFileSync(fullPath, 'utf8');
            const pattern = /backgroundImage:\s*NetworkImage\('https:\/\/i\.pravatar\.cc[^']+'\),?/g;
            const replacement = 'child: const Icon(Icons.person, color: Colors.white),';
            
            if (pattern.test(content)) {
                const newContent = content.replace(pattern, replacement);
                fs.writeFileSync(fullPath, newContent, 'utf8');
                console.log(`Updated ${fullPath}`);
            }
        }
    }
}

removePravatar(__dirname);
console.log('Done!');

File: remove_placeholders.py
import os
import re

def remove_pravatar(directory):
    for root, _, files in os.walk(directory):
        if 'build' in root: continue
        for file in files:
            if file.endswith('.dart'):
                filepath = os.path.join(root, file)
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                # Replace backgroundImage: NetworkImage('https://i.pravatar.cc...')
                # with child: const Icon(Icons.person, color: Colors.white)
                pattern = r"backgroundImage:\s*NetworkImage\('https://i\.pravatar\.cc[^']+'\),*"
                replacement = r"child: const Icon(Icons.person, color: Colors.white),"
                
                new_content = re.sub(pattern, replacement, content)
                
                if new_content != content:
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    print(f"Updated {filepath}")

remove_pravatar(r"c:\Users\ABCD\Desktop\chiu avishkar")
print("Done!")
